### ISSUE TITLE
backend: add tests for default value, update existing tests

### ISSUE THREAD

**Issue by RoboSparrow** [gardners/surveysystem#191]

**Comment by RoboSparrow** [comment]
new answer `flag` field, affected tests

```
grep -nHIrF -- &answer= (in directory: /home/joerg/Workspace/surveysystem/surveysystem/backend/tests)
./001_nextquestionpythonActualLogic.test:50:request 200 addanswer?sessionid=$SESSION&answer=email:test.email@someDomain.org:0:0:0:0:0:0:0:
./answerquestion.test:24:request 200 addanswer?sessionid=$SESSION&answer=question1:Hello+World:0:0:0:0:0:0:0::0
./field_description.test:29:request 200 addanswer?sessionid=$SESSION&answer=question1::5:0:0:0:0:0:0:
./field_description.test:31:request 200 addanswer?sessionid=$SESSION&answer=question2::-5:0:0:0:0:0:0:
./field_unit.test:26:request 200 addanswer?sessionid=$SESSION&answer=question1::28:0:0:0:0:0:0:days
./003_outputFromPythonShouldBeSameAsFromC.test:42:request 200 addanswer?sessionid=$SESSION&answer=email:test.email@someDomain.org:0:0:0:0:0:0:0:
./009_testCheckRecordedData.test:36:request 200 addanswer?sessionid=$SESSION&answer=email:test.email@domain.com:0:0:0:0:0:0:0:
./delanswer-randomly.test:23:request 200 addanswer?sessionid=$SESSION&answer=question1:Answer+1:0:0:0:0:0:0:0:
./delanswer-randomly.test:26:request 200 addanswer?sessionid=$SESSION&answer=question2:Answer+2:0:0:0:0:0:0:0:
./delanswer-randomly.test:29:request 200 addanswer?sessionid=$SESSION&answer=question3:Answer+3:0:0:0:0:0:0:0:
./007_testShouldVerifyDefinitions.test:33:request 200 addanswer?sessionid=$SESSION&answer=email:test.email@domain.com:0:0:0:0:0:0:0:
./QTYPE_MULTICHOICE.test:26:request 200 addanswer?sessionid=$SESSION&answer=multichoice:Marvin,Ford:0:0:0:0:0:0:0:
./sequencedanswering.test:24:request 200 addanswer?sessionid=$SESSION&answer=question1:Hello+World:0:0:0:0:0:0:0:
./sequencedanswering.test:35:request 200 addanswer?sessionid=$SESSION&answer=question2:Another+answer:0:0:0:0:0:0:0:
./sequencedanswering.test:47:request 200 addanswer?sessionid=$SESSION&answer=question3:Last+answer:0:0:0:0:0:0:0:
./004_pythonAlgSequencedAnswer.test:46:request 200 addanswer?sessionid=$SESSION&answer=email:test.email@domain.com:0:0:0:0:0:0:0:
./004_pythonAlgSequencedAnswer.test:58:request 200 addanswer?sessionid=$SESSION&answer=password1:myPaSSw0rD:0:0:0:0:0:0:0:
./004_pythonAlgSequencedAnswer.test:71:request 200 addanswer?sessionid=$SESSION&answer=password2:myPaSSw0rD:0:0:0:0:0:0:0:
./QTYPE_MULTISELECT.test:26:request 200 addanswer?sessionid=$SESSION&answer=multiselect:Marvin,Ford:0:0:0:0:0:0:0:
./QTYPE_SINGLESELECT.test:26:request 200 addanswer?sessionid=$SESSION&answer=singleselect:Ford:0:0:0:0:0:0:0:
./006_testShouldVerifyDefinitions.test:33:request 200 addanswer?sessionid=$SESSION&answer=email:test.email@domain.com:0:0:0:0:0:0:0:
./POST-answerquestion.test:23:request 200 addanswer POST "sessionid=$SESSION&answer=question1:Hello+World:0:0:0:0:0:0:0:"
./delanswer-decremential.test:23:request 200 addanswer?sessionid=$SESSION&answer=question1:Answer+1:0:0:0:0:0:0:0:
./delanswer-decremential.test:26:request 200 addanswer?sessionid=$SESSION&answer=question2:Answer+2:0:0:0:0:0:0:0:
./delanswer-decremential.test:29:request 200 addanswer?sessionid=$SESSION&answer=question3:Answer+3:0:0:0:0:0:0:0:
./character-encoding.test:27:request 200 addanswer?sessionid=$SESSION&answer=characterencoding:Ford:0:0:0:0:0:0:0:
./QTYPE_CHECKBOX.test:26:request 200 addanswer?sessionid=$SESSION&answer=AreYouAwake::28:0:0:0:0:0:0:days
./version2_without_python.test:37:request 200 addanswer?sessionid=$SESSION&answer=question1:Hello+World:0:0:0:0:0:0:0:
./version2_without_python.test:82:request 200 addanswer?sessionid=$SESSION&answer=question1:Hello+World:0:0:0:0:0:0:0:
./005_surveyResultsVerification.test:42:request 200 addanswer?sessionid=$SESSION&answer=email:test.email@domain.com:0:0:0:0:0:0:0:
./005_surveyResultsVerification.test:54:request 200 addanswer?sessionid=$SESSION&answer=password1:myPaSSw0rD:0:0:0:0:0:0:0:
./005_surveyResultsVerification.test:67:request 200 addanswer?sessionid=$SESSION&answer=password2:myPaSSw0rD:0:0:0:0:0:0:0:
./008_testShouldVerifyDefinitions.test:32:request 200 addanswer?sessionid=$SESSION&answer=email:test.email@domain.com:0:0:0:0:0:0:0:
./answernonexistantquestion.test:23:request 400 addanswer?sessionid=$SESSION&answer=nosuchquestion:Hello+World:0:0:0:0:0:0:0:
./QTYPE_SINGLECHOICE.test:26:request 200 addanswer?sessionid=$SESSION&answer=singlechoice:Ford:0:0:0:0:0:0:0:
Search completed with 35 matches.
```

**Comment by RoboSparrow** [comment]
resolved with #200
